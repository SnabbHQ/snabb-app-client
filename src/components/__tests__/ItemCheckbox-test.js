/**
 * # ItemCheckbox-test.js
 *
 * This class tests that the ItemCheckbox renders correctly under
 * numerous conditions
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 *
 */
'use strict'

import 'react-native'
import React from 'react'

import ItemCheckbox from '../ItemCheckbox'

import renderer from 'react/lib/ReactTestRenderer'

/**
 * ## Test
 */
describe('ItemCheckbox', () => {
  /**
   * ### if not disabled and checked, it should display check-square and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if not disabled and checked, it should display check-square and text', () => {
    const props = {
      checked: true,
      text: 'TextShouldDisplay',
      disabled: false
    }
    const tree = renderer.create(<ItemCheckbox {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if not disabled and not checked, it should display square-o and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if not disabled and not checked, it should display square-o and text', () => {
    const props = {
      checked: false,
      text: 'TextShouldDisplay',
      disabled: false
    }
    const tree = renderer.create(<ItemCheckbox {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if disabled and checked, it should display check-square and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if disabled and checked, it should display check-square and text', () => {
    const props = {
      checked: true,
      text: 'TextShouldDisplay',
      disabled: true
    }
    const tree = renderer.create(<ItemCheckbox {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  /**
   * ### if disabled and not checked, it should display square-o and text
   * change the props and call ```testItemCheckbox``` to validate
   */
  it('if disabled and not checked, it should display square-o and text', () => {
    const props = {
      checked: false,
      text: 'TextShouldDisplay',
      disabled: true
    }
    const tree = renderer.create(<ItemCheckbox {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})// describe ItemCheckbox

