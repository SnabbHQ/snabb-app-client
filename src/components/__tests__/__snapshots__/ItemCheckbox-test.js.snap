exports[`ItemCheckbox if disabled and checked, it should display check-square and text 1`] = `
<View
  style={undefined}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
       
      TextShouldDisplay
    </Text>
  </View>
</View>
`;

exports[`ItemCheckbox if disabled and not checked, it should display square-o and text 1`] = `
<View
  style={undefined}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
       
      TextShouldDisplay
    </Text>
  </View>
</View>
`;

exports[`ItemCheckbox if not disabled and checked, it should display check-square and text 1`] = `
<View
  style={undefined}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </View>
</View>
`;

exports[`ItemCheckbox if not disabled and not checked, it should display square-o and text 1`] = `
<View
  style={undefined}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 20,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
         
        TextShouldDisplay
      </Text>
    </View>
  </View>
</View>
`;
